### PR TITLE
[WIP] :book: Add Release Writer to release team roles

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add a Release Writer role to the the release tasks doc. This role has become de facto required as part of the release process. This change formalizes the role and defines the responsibilities and tasks of someone who is the designated Release Writer for a release.

### CODE REVIEW

**Comment by killianmuldoon** [comment]
/hold 

For feedback from the community meeting.

**Comment by elmiko** [comment]
we are discussing this during the community meeting on 12 July, there are some questions about this process and the name. cc @CecileRobertMichon for more comments.

**Comment by CecileRobertMichon** [comment]
TLDR from the office hours discussion:

- general consensus that having a maintainer available for help and support during release days is a good idea, +1 to formalize this
- we should still look into giving the release team all the permissions they need to be able to release end to end without needing maintainers' help (this might mean setting up more automation, setting up a release manager team [similar to k8s](https://github.com/orgs/kubernetes/teams/release-managers) with a [custom github role](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/managing-custom-repository-roles-for-an-organization), etc.)
- Preference around naming this role something like "on-call maintainer" or "supporting maintainer" so it's clear this role has to be filled by a project maintainer as opposed to something generic like "release writer"

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -10,43 +10,47 @@ This document details the responsibilities and tasks for each role in the releas
...
+    - [Continuously: Communicate key dates to the community](#continuously-communicate-key-dates-to-the-community)
+    - [Communicate beta to providers](#communicate-beta-to-providers)
+- [CI Signal/Bug Triage/Automation Manager](#ci-signalbug-triageautomation-manager)
+  - [Responsibilities](#responsibilities-2)
+  - [Tasks](#tasks-2)
+    - [Setup jobs and dashboards for a new release branch](#setup-jobs-and-dashboards-for-a-new-release-branch)
+    - [[Continuously] Monitor CI signal](#continuously-monitor-ci-signal)
+    - [[Continuously] Reduce the amount of flaky tests](#continuously-reduce-the-amount-of-flaky-tests)
+    - [[Continuously] Bug triage](#continuously-bug-triage)
+- [Release Writer](#release-writer)
```
Comment (fabriziopandini) [inline docs/release/release-tasks.md:49]: What about
```suggestion
- [Mantainer aid](#release-writer)
```

**Review by fabriziopandini — COMMENTED** [review]
Only a nitpick on the role name otherwise +1 for me

File: docs/release/release-tasks.md (hunk 1 above)
Comment (sbueringer) [reply docs/release/release-tasks.md:49]: We could also simplify and just call it Maintainer directly :) 

(I'm fine either way)

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -10,43 +10,47 @@ This document details the responsibilities and tasks for each role in the releas
...
+  - [Responsibilities](#responsibilities-3)
+  - [Tasks](#tasks-3)
```
Comment (sbueringer) [inline docs/release/release-tasks.md:51]: Maybe we should also update the issue template accordingly, not sure

(maybe we should just write "Release Lead/Maintainer" at the tasks which are "shared")

File: docs/release/release-tasks.md [hunk 3]
```diff
@@ -491,3 +472,52 @@ and add them to the milestone of the current release.
...
 We probably have to figure out some details about the overlap between the bug triage task here, release leads
 and Cluster API maintainers.
+
+## Release Writer
+
+The Release Writer must be a person with write access to the Cluster API repo. They can hold another role in the release team. The Release Writer need only be involved in the release as required on days when releases are cut. They are not expected to take part in release team meetings or other activities, but should feel free to do so.
+
+### Responsibilities
+  * Create new release branches.
+  * Write tags for releases.
```
Comment (sbueringer) [inline docs/release/release-tasks.md:482]:
```suggestion
  * Create tags for releases.
```
nit

File: docs/release/release-tasks.md [hunk 4]
```diff
@@ -491,3 +472,52 @@ and add them to the milestone of the current release.
...
+  * Publish the release. 
+  * Ensure a substitute is nominated to cut a release if they are not available.
+
+### Tasks
+
+#### Prepare main branch for development of the new release
+Create the release branch locally based on the latest commit on main and push it.
+   ```bash
```
Comment (sbueringer) [inline docs/release/release-tasks.md:490]: nit: indent looks a bit off

(similar below)

File: docs/release/release-tasks.md [hunk 5]
```diff
@@ -491,3 +472,52 @@ and add them to the milestone of the current release.
...
+   # Warning: The test tag MUST NOT be an annotated tag.
+   git tag -s -a ${RELEASE_TAG} -m ${RELEASE_TAG}
+   git tag test/${RELEASE_TAG}
+
+   # Push tags
+   # Note: `upstream` must be the remote pointing to `github.com/kubernetes-sigs/cluster-api`.
+   git push upstream ${RELEASE_TAG}
+   git push upstream test/${RELEASE_TAG}
+   ```
+   **Note**:
```
Comment (sbueringer) [inline docs/release/release-tasks.md:519]:
```suggestion
```

**Review by sbueringer — COMMENTED** [review]
Just nits

File: docs/release/release-team.md [hunk 6]
```diff
@@ -49,8 +67,8 @@ As noted above, making changes to  the CAPI release cadence is out of scope for
...
 - **CI Signal/Bug Triage/Automation Manager**: Assumes the responsibility of the quality gate for the release and makes sure blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
-- **Shadow**: Any Release Team member may select one or more mentees to shadow the release process in order to help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
-
+- **Team member**: Any Release Team lead or manager may select one or more additional members to help with their the release tasks. These team members will help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
+- **Release Writer**: responsible for tasks which require write access to the Cluster API repo including creating release tags and creating a release branch. This role must be filled by someone on the [`cluster-api-maintainers` list](https://github.com/kubernetes-sigs/cluster-api/blob/5999d1679509d4f3b09d50f394c43bc0f8d33c94/OWNERS_ALIASES#L21-L27).
```
Comment (sbueringer) [inline docs/release/release-team.md:71]:
```suggestion
- **Release Writer**: Responsible for tasks which require write access to the Cluster API repo including creating release tags and creating a release branch. This role must be filled by someone on the [`cluster-api-maintainers` list](https://github.com/kubernetes-sigs/cluster-api/blob/5999d1679509d4f3b09d50f394c43bc0f8d33c94/OWNERS_ALIASES#L21-L27).
```
nit

I think we should link the file on main instead, but then we should drop the lines in the link

File: docs/release/release-tasks.md (hunk 1 above)
Comment (killianmuldoon) [reply docs/release/release-tasks.md:49]: Called it maintainer for now - I'm fine with any name that people are happy with.

**Comment by sbueringer** [comment]
/lgtm
You'll need a rebase now though

**Comment by sbueringer** [comment]
/lgtm

I'll give others a chance to review. Would be good if we can get this merged with the next week or so. So we have it ready for the next cycle.

**Comment by killianmuldoon** [comment]
/hold cancel

@sbueringer WDYT about merging this as-is? If there's further feedback we can update before the next release team comes onborad.

**Comment by sbueringer** [comment]
Sounds good.

If there is any further feedback, please let us know and we can do a follow-up PR

/approve

File: docs/release/releases/release-1.6.md [hunk 7]
```diff
@@ -32,4 +32,5 @@ After the `.0` release monthly patch release will be created.
 |-------------------------------------------|--------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | Release Lead                              | TBD | TBD   |
 | Communications/Docs/Release Notes Manager | TBD | TBD   |
-| CI Signal/Bug Triage/Automation Manager   | TBD | TBD   |
\ No newline at end of file
+| CI Signal/Bug Triage/Automation Manager   | TBD | TBD   |
+| Release Writer                            | TBD | TBD   |
```
Comment (CecileRobertMichon) [inline docs/release/releases/release-1.6.md:36]: sorry for the late review! this should also be renamed to `Maintainer` 

otherwise LGTM

Comment (killianmuldoon) [reply docs/release/releases/release-1.6.md:36]: Thanks for catching that - opened a PR at https://github.com/kubernetes-sigs/cluster-api/pull/9013
